### PR TITLE
Fix failed tests

### PR DESCRIPTION
**Pull request by zhengbli**
Due to the previous error in test script, we have 3 tests are actually failed currently. This PR fixes these failed tests and added a new test for PR #48 which was merged when the CI wasn't working properly.

### CODE REVIEW

**Comment by zhengbli** [comment]
@DanielRosenwasser @stkb comments?

File: TypeScript.YAML-tmLanguage [hunk 1]
```diff
@@ -410,6 +402,39 @@ repository:
     - include: '#switch-case'
     - include: '#control-statement'
 
+  switch-variable:
```
Comment (DanielRosenwasser) [inline TypeScript.YAML-tmLanguage]: switch-expression

File: TypeScript.YAML-tmLanguage [hunk 2]
```diff
@@ -410,6 +402,39 @@ repository:
...
+    - include: '#expression'
+
+  switch-block:
+    name: switch-block.expr.ts
+    begin: '{'
+    end: (?=\})
+    patterns:
+    - include: '#expression'
+
+  switch-case:
```
Comment (DanielRosenwasser) [inline TypeScript.YAML-tmLanguage]: case-clause

File: TypeScript.YAML-tmLanguage [hunk 3]
```diff
@@ -410,6 +402,39 @@ repository:
...
+    name: case.expr.ts
+    begin: '(?<!\.)\b(case|default)\b'
+    beginCaptures:
+      '1': { name: keyword.control.ts }
+    end: ':'
+    patterns:
+    - include: '#expression'
+
+  switch-expression:
```
Comment (DanielRosenwasser) [inline TypeScript.YAML-tmLanguage]: switch-statement

File: TypeScript.YAML-tmLanguage [hunk 4]
```diff
@@ -169,7 +160,7 @@ repository:
       '5': { name: keyword.operator.ts }
       '6': { name: entity.name.function.ts }
       '7': { name: keyword.operator.ts }
-    end: '(?=\}|;|,|$)|(?<=\})'
```
Comment (DanielRosenwasser) [inline TypeScript.YAML-tmLanguage]: Hmm, the thing is, a method declaration might not have a body and might just need on a newline. For instance,

```ts
abstract class C {
    abstract foo()
    abstract bar()
}
```

or with a method signature:

```ts
interface I {
    foo(): number
    bar(): string;
}

Comment (DanielRosenwasser) [inline TypeScript.YAML-tmLanguage]: Why was this changed? For braces on the next line?

Comment (zhengbli) [inline TypeScript.YAML-tmLanguage]: This change because the current definition breaks the following case:
```ts
class Test {
	acid(trip: string | {}, buttweasel: string) 
	{ // here
	}
}
```
because it thought the method definition ends at new line

Comment (DanielRosenwasser) [inline TypeScript.YAML-tmLanguage]: Hm - I guess there's not really a problem, since you'll just highlight as a block anyway. This should be fine

Comment (zhengbli) [inline TypeScript.YAML-tmLanguage]: Not it would not be fine. The block could be reorganized as an object body, and it would be hard to differentiate these two cases:
```ts
abstract class C {
    abstract foo() :
    {// object return type
    }
}
```
and 
```ts
abstract class C {
    abstract foo()
    {// method implementation
    }
}
```

**Comment by zhengbli** [comment]
@DanielRosenwasser any further comments? Added the two cases you mentioned to the test.

**Comment by DanielRosenwasser** [comment]
:+1:

File: TypeScript.YAML-tmLanguage (hunk 4 above)
Comment (DanielRosenwasser) [inline TypeScript.YAML-tmLanguage]: Sorry, by fine I meant your changes should be fine, not the code as it was originally written.
